### PR TITLE
task4_security

### PR DESCRIPTION
**Pull request by egor-geller**

### CODE REVIEW

File: service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -40,8 +56,49 @@ public UserDto findUserByIdService(Long id) {
...
+
+        String encodedPassword = encoder.encode(user.getPassword());
+        user.setPassword(encodedPassword);
+
+        User savedUser = userRepository.save(user);
+        Optional<User> userById = userRepository.findById(savedUser.getId());
+        if (userById.isEmpty()) {
+            throw new EntityNotFoundException(String.valueOf(savedUser.getId()));
+        }
+        return userServiceMapper.convertUserToDto(userById.get());
```
Comment (iyanushkevich) [inline service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java]: Why make another call to the database? Get the user object as a result of saving it to the database and return it.

File: service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -40,8 +56,49 @@ public UserDto findUserByIdService(Long id) {
         return userServiceMapper.convertUserToDto(user);
     }
 
+    @Transactional
     @Override
-    public Long count(){
+    public UserDto signup(UserDto userDto) {
+        User user = userServiceMapper.convertUserFromDto(userDto);
+
+        List<ValidationError> validationErrors = userValidator.validate(user);
```
Comment (iyanushkevich) [inline service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java]: Is it advisable to do the conversion from DTO, and then validate? Why can't we immediately validate the object that we received from the request, and then, if necessary, validate it?

File: service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java [hunk 3]
```diff
@@ -40,8 +56,49 @@ public UserDto findUserByIdService(Long id) {
         return userServiceMapper.convertUserToDto(user);
     }
 
+    @Transactional
     @Override
-    public Long count(){
```
Comment (iyanushkevich) [inline service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java:44]: I saw you declared this method in the interface. If it is not necessary - delete it

File: service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java [hunk 4]
```diff
@@ -40,8 +56,49 @@ public UserDto findUserByIdService(Long id) {
...
+            throw new EntityNotFoundException(String.valueOf(savedUser.getId()));
+        }
+        return userServiceMapper.convertUserToDto(userById.get());
+    }
+
+    @Override
+    public UserDto login(UserDto userDto) {
+        String username = userDto.getUsername();
+        String password = userDto.getPassword();
+
```
Comment (iyanushkevich) [inline service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java]: I would not recommend getting used to leaving blank lines. This clearly violates the Java Code Convention. But in some cases it improves the readability of the code. In your case, there is definitely no such need

File: service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java [hunk 5]
```diff
@@ -40,8 +56,49 @@ public UserDto findUserByIdService(Long id) {
...
+        User savedUser = userRepository.save(user);
+        Optional<User> userById = userRepository.findById(savedUser.getId());
+        if (userById.isEmpty()) {
+            throw new EntityNotFoundException(String.valueOf(savedUser.getId()));
+        }
+        return userServiceMapper.convertUserToDto(userById.get());
+    }
+
+    @Override
+    public UserDto login(UserDto userDto) {
```
Comment (iyanushkevich) [inline service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java]: Why the entity of the whole user is used for authentication. It seems to me that it would be better if you create a separate AuthenticationDto

File: service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java [hunk 6]
```diff
@@ -40,8 +56,49 @@ public UserDto findUserByIdService(Long id) {
...
+        String username = userDto.getUsername();
+        String password = userDto.getPassword();
+
+        User user = userRepository.findByUsername(username)
+                .orElseThrow(() -> new BadCredentialsException(StringUtils.EMPTY));
```
Comment (iyanushkevich) [inline service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java]: Passing empty lines is too rude. Think about how to improve the message in case of incorrect authentication data

File: service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java [hunk 7]
```diff
@@ -40,8 +56,49 @@ public UserDto findUserByIdService(Long id) {
...
+
+        if (!encoder.matches(password, user.getPassword())) {
+            throw new BadCredentialsException(StringUtils.EMPTY);
```
Comment (iyanushkevich) [inline service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java]: Similarly with the comment above

File: service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java [hunk 8]
```diff
@@ -40,8 +56,49 @@ public UserDto findUserByIdService(Long id) {
...
+        }
+
+        return userDto;
+    }
+
+    @Override
+    public Long count() {
```
Comment (iyanushkevich) [inline service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java:46]: Why do we need this method?

File: service/src/main/java/com/epam/esm/validator/UserValidator.java [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+            validationErrors.add(INVALID_USERNAME);
+        }
+
+        boolean passwordIsValid = validatePassword(password);
+        if (!passwordIsValid) {
+            validationErrors.add(INVALID_PASSWORD);
+        }
+
+        return validationErrors;
+    }
```
Comment (iyanushkevich) [inline service/src/main/java/com/epam/esm/validator/UserValidator.java]: If you have a million user fields? And there will be an error when validating the first of them, will you continue to check all the others? Given that you're throwing an exception anyway.

File: service/src/main/java/com/epam/esm/validator/UserValidator.java [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+
+    private boolean validateUsername(String username) {
+        return Pattern.matches(USERNAME_REGEX, username);
+    }
+
+    private boolean validatePassword(String password) {
+        return Pattern.matches(PASSWORD_REGEX, password);
+    }
```
Comment (iyanushkevich) [inline service/src/main/java/com/epam/esm/validator/UserValidator.java]: It can be combined into one method that will accept a validation field and a regular expression as input

File: service/src/main/java/com/epam/esm/validator/ValidationError.java [hunk 11]
```diff
@@ -0,0 +1,10 @@
+package com.epam.esm.validator;
+
+public enum ValidationError {
+    INVALID_NAME,
+    INVALID_DESCRIPTION,
+    INVALID_PRICE,
+    INVALID_DURATION,
+    INVALID_USERNAME,
+    INVALID_PASSWORD
+}
```
Comment (iyanushkevich) [inline service/src/main/java/com/epam/esm/validator/ValidationError.java:10]: I don't see the need to create an enumeration.

File: service/src/test/java/com/epam/esm/service/CertificateServiceImplTest.java [hunk 12]
```diff
@@ -67,184 +67,184 @@ static void setUp() {
...
 
-        when(certificateRepository.find(provideSearchParameters())).thenReturn(certificateList);
-        when(tagRepository.findByCertificateId(anyLong())).thenReturn(createTagList());
+            when(certificateRepository.find(provideSearchParameters())).thenReturn(certificateList);
+            when(tagRepository.findByCertificateId(anyLong())).thenReturn(createTagList());
 
-        List<CertificateDto> actualDtoList = certificateService
-                .findCertificateByCriteria(provideSearchParameters());
+            List<CertificateDto> actualDtoList = certificateService
+                    .findCertificateByCriteria(provideSearchParameters());
```
Comment (iyanushkevich) [inline service/src/test/java/com/epam/esm/service/CertificateServiceImplTest.java:81]: Avoid commented-out code

File: service/src/test/java/com/epam/esm/service/CertificateServiceImplTest.java [hunk 13]
```diff
@@ -67,184 +67,184 @@ static void setUp() {
...
-        int expectedInteractions = 1;
-        verify(certificateRepository, times(expectedInteractions)).find(provideSearchParameters());
+            int expectedInteractions = 1;
+            verify(certificateRepository, times(expectedInteractions)).find(provideSearchParameters());
 
-        assertEquals(certificateDtoList.toString(), actualDtoList.toString());
-    }
-
-    @Test
-    void findCertificateByIdTest() {
```
Comment (iyanushkevich) [inline service/src/test/java/com/epam/esm/service/CertificateServiceImplTest.java:91]: Why did you delete half of the tests?

File: service/src/test/java/com/epam/esm/service/CertificateServiceImplTest.java [hunk 14]
```diff
@@ -67,184 +67,184 @@ static void setUp() {
...
-        CertificateDto updatedCertificateDto = createCertificateDto();
-        stringList.add("");
-        updatedCertificateDto.setTagList(stringList);
+            Assertions.assertThrows(InvalidEntityException.class, () -> certificateService.create(certificateDto));
+        }
 
-        Assertions.assertThrows(InvalidEntityException.class, () -> certificateService.update(updatedCertificateDto));
-    }
+        @Test
+        void createAlreadyExistsCertificateTest() {
```
Comment (iyanushkevich) [inline service/src/test/java/com/epam/esm/service/CertificateServiceImplTest.java:137]: https://stackoverflow.com/questions/155436/unit-test-naming-best-practices
Edit the names of your test methods according to the accepted norms

File: web/src/main/java/com/epam/esm/security/SecurityConfig.java [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http.csrf().disable()
+                .httpBasic().disable()
+                .sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS)
+                .and()
+                .authorizeRequests()
+                .antMatchers(SIGNUP_ENDPOINT, LOGIN_ENDPOINT).permitAll()
+                .antMatchers(HttpMethod.GET, MAIN_ENTITY_ENDPOINT).permitAll()
+                .antMatchers(GENERATE_ENDPOINT).permitAll()
```
Comment (iyanushkevich) [inline web/src/main/java/com/epam/esm/security/SecurityConfig.java]: If for some method to work with the certificate I need to restrict access to the user? Do I have to add something else here, add each endpoint for each method and specify it?
Read about SPEL. And use the PreAuthorize, PostAuthorize annotations for methods in controller classes

File: web/src/main/java/com/epam/esm/security/SecurityConfig.java [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+                .anyRequest().authenticated()
+                .and()
+                .exceptionHandling()
+                .authenticationEntryPoint(authenticationEntryPoint);
+    }
+
+    @Bean
+    public PasswordEncoder passwordEncoder() {
+        return new BCryptPasswordEncoder();
+    }
```
Comment (iyanushkevich) [inline web/src/main/java/com/epam/esm/security/SecurityConfig.java:58]: This bean is used in the service, why is it in the configuration for the web module?

File: web/src/main/java/com/epam/esm/utils/ResponseUtil.java [hunk 17]
```diff
@@ -0,0 +1,43 @@
...
+
+    public Map<String, Object> buildErrorResponseMap(String errorMessage) {
+        Map<String, Object> map = new HashMap<>();
+        map.put(ERROR_MESSAGE, errorMessage);
+
+        return map;
+    }
+
+    public ResponseEntity<Object> buildErrorResponseEntity(HttpStatus status, String errorMessage) {
+        Map<String, Object> body = buildErrorResponseMap(errorMessage);
```
Comment (iyanushkevich) [inline web/src/main/java/com/epam/esm/utils/ResponseUtil.java]: A separate map will be created for each call of this method. For what?

File: web/src/main/java/com/epam/esm/utils/ResponseUtil.java [hunk 18]
```diff
@@ -0,0 +1,43 @@
...
+        this.messageSource = messageSource;
+    }
+
+    public String getErrorMessage(String errorMessageName) {
+        Locale locale = LocaleContextHolder.getLocale();
+        return messageSource.getMessage(errorMessageName, null, locale);
+    }
+
+    public Map<String, Object> buildErrorResponseMap(String errorMessage) {
+        Map<String, Object> map = new HashMap<>();
```
Comment (iyanushkevich) [inline web/src/main/java/com/epam/esm/utils/ResponseUtil.java]: A separate map will be created for each call of this method. For what?

File: web/src/main/java/com/epam/esm/security/SecurityConfig.java [hunk 19]
```diff
@@ -0,0 +1,53 @@
...
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.AuthenticationEntryPoint;
+
+@Configuration
+@EnableWebSecurity
+@EnableGlobalMethodSecurity(prePostEnabled = true)
```
Comment (iyanushkevich) [inline web/src/main/java/com/epam/esm/security/SecurityConfig.java:23]: You specified prePostEnabled,  based on your implementation why do you need this at this stage? Do you understand what this is used for?

File: service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java (hunk 6 above)
Comment (egor-geller) [reply service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java]: fixed

File: service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java (hunk 7 above)
Comment (egor-geller) [reply service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java]: fixed

File: service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java (hunk 8 above)
Comment (egor-geller) [reply service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java:46]: Its for pagination purposes

File: service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java (hunk 6 above)
Comment (egor-geller) [reply service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java]: In second thought, this exception doesn't have an empty constructor, a meaningful message (if the exception is thrown) is processed by a CustomExceptionHandler

File: service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java (hunk 7 above)
Comment (egor-geller) [reply service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java]: In second thought, this exception doesn't have an empty constructor, a meaningful message (if the exception is thrown) is processed by a CustomExceptionHandler

File: web/src/main/java/com/epam/esm/security/SecurityConfig.java (hunk 16 above)
Comment (egor-geller) [reply web/src/main/java/com/epam/esm/security/SecurityConfig.java:58]: Should I create another config in service only for this bean?

File: service/src/main/java/com/epam/esm/validator/UserValidator.java (hunk 10 above)
Comment (egor-geller) [reply service/src/main/java/com/epam/esm/validator/UserValidator.java]: fixed

File: service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java (hunk 1 above)
Comment (egor-geller) [reply service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java]: fixed

File: service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java (hunk 2 above)
Comment (egor-geller) [reply service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java]: fixed

File: service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java (hunk 3 above)
Comment (egor-geller) [reply service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java:44]: Its for pagination purposes

File: web/src/main/java/com/epam/esm/controller/UserController.java [hunk 20]
```diff
@@ -77,11 +90,49 @@ public UserController(UserService userService,
...
+    /**
+     * Create a new user.
+     * Access is allowed to everyone.
+     *
+     * @param userDto {@link UserDto} instance
+     * @throws InvalidEntityException in case when passed DTO object contains invalid data
+     * @throws EntityAlreadyExistsException in case when user with specified username already exists
+     * @return JSON {@link ResponseEntity} with {@link HateoasModel} object that contains {@link UserDto} object
+     */
+    @PostMapping("/signup")
```
Comment (nickaden) [inline web/src/main/java/com/epam/esm/controller/UserController.java]: I think that UserController shouldn't have Authorization responsibility. Let's extract this logic to new, AuthenticationController

File: web/src/main/java/com/epam/esm/controller/UserController.java [hunk 21]
```diff
@@ -77,11 +90,49 @@ public UserController(UserService userService,
...
+    public ResponseEntity<HateoasModel<UserDto>> signup(@RequestBody UserDto userDto) {
+        UserDto signup = userService.signup(userDto);
+        HateoasModel<UserDto> model = new HateoasModel<>(signup);
+        HateoasModel<UserDto> build = model.build(modelHateoasProvider, signup, 1L);
```
Comment (nickaden) [inline web/src/main/java/com/epam/esm/controller/UserController.java]: signup is not REST endpoint, so you don't need use HatEOAS here

File: web/src/main/java/com/epam/esm/security/SecurityConfig.java [hunk 22]
```diff
@@ -0,0 +1,60 @@
...
+
+    private final AuthenticationEntryPoint authenticationEntryPoint;
+
+    @Autowired
+    public SecurityConfig(AuthenticationEntryPoint authenticationEntryPoint) {
+        this.authenticationEntryPoint = authenticationEntryPoint;
+    }
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
```
Comment (nickaden) [inline web/src/main/java/com/epam/esm/security/SecurityConfig.java:45]: Do you really need this since you are using method security?
https://www.baeldung.com/spring-security-method-security

File: web/src/main/java/com/epam/esm/controller/UserController.java (hunk 20 above)
Comment (egor-geller) [reply web/src/main/java/com/epam/esm/controller/UserController.java]: fixed

File: web/src/main/java/com/epam/esm/controller/UserController.java (hunk 21 above)
Comment (egor-geller) [reply web/src/main/java/com/epam/esm/controller/UserController.java]: fixed

File: web/src/main/java/com/epam/esm/security/SecurityConfig.java (hunk 22 above)
Comment (egor-geller) [reply web/src/main/java/com/epam/esm/security/SecurityConfig.java:45]: fixed

File: web/src/main/java/com/epam/esm/security/SecurityConfig.java (hunk 19 above)
Comment (egor-geller) [reply web/src/main/java/com/epam/esm/security/SecurityConfig.java:23]: fixed

File: web/src/main/java/com/epam/esm/security/SecurityConfig.java (hunk 15 above)
Comment (egor-geller) [reply web/src/main/java/com/epam/esm/security/SecurityConfig.java]: fixed

File: service/src/main/java/com/epam/esm/service/UserService.java [hunk 23]
```diff
@@ -29,5 +34,24 @@
...
 
+    /**
+     * Create a new user.
+     *
+     * @param userDto {@link UserDto} instance
+     * @throws InvalidEntityException in case when passed DTO object contains invalid data
+     * @throws EntityAlreadyExistsException in case when user with specified username already exists
+     * @return {@link TokenDto} object
+     */
+    TokenDto signup(UserDto userDto);
```
Comment (nickaden) [inline service/src/main/java/com/epam/esm/service/UserService.java]: signUp and login is not in UserService scope. Please create another service with UserService as dependency

File: service/src/main/java/com/epam/esm/service/impl/MyUserDetailsService.java [hunk 24]
```diff
@@ -0,0 +1,39 @@
...
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Optional;
+
+@Service
+public class MyUserDetailsService implements UserDetailsService {
```
Comment (nickaden) [inline service/src/main/java/com/epam/esm/service/impl/MyUserDetailsService.java]: MyUserDetailsService - do you think it is good name for business?

File: web/src/main/java/com/epam/esm/controller/GenerateData.java [hunk 25]
```diff
@@ -1,6 +1,7 @@
 package com.epam.esm.controller;
```
Comment (nickaden) [inline web/src/main/java/com/epam/esm/controller/GenerateData.java]: If you're using this class to generate data in DB, please move it to separate module (such as `uat` - 'user acceptance tests')

File: web/src/main/java/com/epam/esm/controller/OrderController.java [hunk 26]
```diff
@@ -108,6 +111,7 @@ public OrderController(OrderService orderService,
      * @return JSON {@link ResponseEntity} object that contains list {@link ListHateoasModel} of {@link OrderDto}
      */
     @GetMapping("/user/{id}")
+    @PreAuthorize("hasAuthority('ADMIN') or hasAuthority('USER')")
```
Comment (nickaden) [inline web/src/main/java/com/epam/esm/controller/OrderController.java]: May user see order of another user passing another user's ID?

File: web/src/main/java/com/epam/esm/filter/CustomAuthenticationFilter.java [hunk 27]
```diff
@@ -0,0 +1,72 @@
...
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+public class CustomAuthenticationFilter extends UsernamePasswordAuthenticationFilter {
```
Comment (nickaden) [inline web/src/main/java/com/epam/esm/filter/CustomAuthenticationFilter.java]: You missed Authentication and Authorization definitions

File: web/src/main/java/com/epam/esm/security/AppAuthenticationEntryPoint.java [hunk 28]
```diff
@@ -0,0 +1,61 @@
...
+    @Autowired
+    public AppAuthenticationEntryPoint(ResourceBundleMessageSource bundleMessageSource) {
+        this.bundleMessageSource = bundleMessageSource;
+    }
+
+    @Override
+    public void commence(HttpServletRequest request,
+                         HttpServletResponse response,
+                         AuthenticationException authException) throws IOException {
+        String errorMessage = getErrorMessage();
```
Comment (nickaden) [inline web/src/main/java/com/epam/esm/security/AppAuthenticationEntryPoint.java]: I don't like that you create exceptions in two places - in ControllerAdvice and here. Let's inject controller advice as dependency and invoke dedicated method in `commence`

File: web/src/main/java/com/epam/esm/filter/CustomAuthenticationFilter.java [hunk 29]
```diff
@@ -0,0 +1,72 @@
...
+        return authenticationManager.authenticate(authenticationToken);
+    }
+
+    @Override
+    protected void successfulAuthentication(HttpServletRequest request, HttpServletResponse response, FilterChain chain, Authentication authResult) throws IOException, ServletException {
+        User user = (User) authResult.getPrincipal();
+        Algorithm algorithm = Algorithm.HMAC256("secret".getBytes());
+        String accessToken = JWT.create()
+                .withSubject(user.getUsername())
+                .withExpiresAt(new Date(System.currentTimeMillis() + 10 * 60 * 1000))
```
Comment (nickaden) [inline web/src/main/java/com/epam/esm/filter/CustomAuthenticationFilter.java]: Let's configure it with applicaiton.properties

File: web/src/main/java/com/epam/esm/security/AppAuthenticationEntryPoint.java (hunk 28 above)
Comment (egor-geller) [reply web/src/main/java/com/epam/esm/security/AppAuthenticationEntryPoint.java]: fixed

File: service/src/main/java/com/epam/esm/service/impl/MyUserDetailsService.java (hunk 24 above)
Comment (egor-geller) [reply service/src/main/java/com/epam/esm/service/impl/MyUserDetailsService.java]: fixed

File: service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java (hunk 5 above)
Comment (egor-geller) [reply service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java]: fixed

File: web/src/main/java/com/epam/esm/utils/ResponseUtil.java (hunk 17 above)
Comment (egor-geller) [reply web/src/main/java/com/epam/esm/utils/ResponseUtil.java]: fixed

File: web/src/main/java/com/epam/esm/utils/ResponseUtil.java (hunk 18 above)
Comment (egor-geller) [reply web/src/main/java/com/epam/esm/utils/ResponseUtil.java]: fixed

File: service/src/main/java/com/epam/esm/service/UserService.java (hunk 23 above)
Comment (egor-geller) [reply service/src/main/java/com/epam/esm/service/UserService.java]: fixed

File: web/src/main/java/com/epam/esm/controller/GenerateData.java (hunk 25 above)
Comment (egor-geller) [reply web/src/main/java/com/epam/esm/controller/GenerateData.java]: fixed

File: web/src/main/java/com/epam/esm/filter/CustomAuthenticationFilter.java (hunk 29 above)
Comment (egor-geller) [reply web/src/main/java/com/epam/esm/filter/CustomAuthenticationFilter.java]: fixed
